bundles/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/data/HueSettings.java

```diff
@@ -223,6 +223,10 @@ protected int count() {
 			return dataMap.size();
 		}
 
+		protected Set<String> getKeys(){
+			return dataMap.keySet();
+		}
```

Comment:
Actually the Philips API is not clear about a type for the ID. Since the usage of the ID in the rest API is always in a string context (and the integer ID is alway converted into string), my next suggestion for a refactoring would have been a complete usage of String for the ID type.
In the ESH the lights ID is also a String.